app/repositories/SessionRepository.scala

```diff
@@ -155,13 +157,16 @@ class DefaultSessionRepository @Inject() (config: AppConfig, component: Reactive
       }
   }
 
-  def saveUserAnswerAndLabels(key: String, url: String, answer: String, labels: Seq[Label]): Future[RequestOutcome[Unit]] =
+  def saveFormPageState(key: String, url: String, answer: String, labels: Seq[Label], stack: List[FlowStage]): Future[RequestOutcome[Unit]] =
     findAndUpdate(
       Json.obj("_id" -> key),
       Json.obj(
         "$set" -> Json.obj(
-          (List(toFieldPair(ttlExpiryFieldName, Json.obj(toFieldPair("$date", Instant.now().toEpochMilli))),
-                toFieldPair(s"answers.$url", answer)) ++ labels.map(l => toFieldPair(s"labels.${l.name}", l))).toArray: _*
+          (List(
+            toFieldPair(ttlExpiryFieldName, Json.obj(toFieldPair("$date", Instant.now().toEpochMilli))),
+            toFieldPair(s"answers.$url", answer)) ++
+            labels.map(l => toFieldPair(s"labels.${l.name}", l)) ++
+            stack.map(e => toFieldPair(s"flowStack", e))).toArray: _*
```

Comment:
Each element in the FlowStage list will have the field name "flowStack". Is that Ok?

Reply:
not quite, but updated to persist correctly